src/engine/sync/internalclock.cpp

```diff
@@ -36,7 +36,7 @@ InternalClock::InternalClock(const char* pGroup, SyncableListener* pEngineSync)
         new ControlPushButton(ConfigKey(pGroup, "sync_master")));
     m_pSyncMasterEnabled->setButtonMode(ControlPushButton::TOGGLE);
     m_pSyncMasterEnabled->connectValueChangeRequest(
-        this, SLOT(slotSyncMasterEnabledChangeRequest(double)),
+        this, [=](double value){slotSyncMasterEnabledChangeRequest(value);},
```

Comment:
Why a lambda here (and in other diffs) instead of `&InternalClock::slotSyncMasterEnabledChangeRequest`?

Reply:
Because that doesn't compile:
```
src/engine/loopingcontrol.cpp: In constructor ‘LoopingControl::LoopingControl(QString, UserSettingsPointer)’:
src/engine/loopingcontrol.cpp:131:81: error: no matching function for call to ‘ControlObject::connectValueChangeRequest(LoopingControl*, void (LoopingControl::*)(double), Qt::ConnectionType)’
          &LoopingControl::slotBeatLoopSizeChangeRequest, Qt::DirectConnection);
                                                                              ^

In file included from src/engine/loopingcontrol.cpp:7:
src/control/controlobject.h:154:10: note: candidate: ‘bool ControlObject::connectValueChangeRequest(const QObject*, std::function<void(double)>, Qt::ConnectionType)’
     bool connectValueChangeRequest(const QObject* receiver,
          ^~~~~~~~~~~~~~~~~~~~~~~~~
src/control/controlobject.h:154:10: note:   no known conversion for argument 2 from ‘void (LoopingControl::*)(double)’ to ‘std::function<void(double)>’
```

Reply:
Lambdas seem to be the easiest way to use a class member function as a std::function.

Reply:
std::bind also works, but IMO the `_1` placeholder thing it requires is weird and I'd prefer to use a lambda.

Reply:
Ah that's because you're making connectValueChanged take a std::function, while pointer to member is its own type and not convertible to std::function. 

You can get around this by making connectValueChanged a template so you aren't writing down the type of method, then you can pass through method to  connect() and the appropriate version of connect will get called based on whether it's pointer to member or a functor.